### PR TITLE
[GPU] Allow U8 data type for SDPA attention mask

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - Some LLMs, such as Qwen2-0.5B, the attention mask input type of IR converted from the latest transformers version(>=4.53.0) has been changed from floating-point to boolean
 - Boolean is converted to U8 by `ConvertPrecision` transformation. But this is not allowed by op validation now
 - So allow U8 data type for SDPA attention mask and relaxes op validation's condition for U8

### Tickets:
 - 171598, 171516

### CODE REVIEW

File: src/core/src/op/scaled_dot_product_attention.cpp [hunk 1]
```diff
@@ -44,14 +44,14 @@ void op::v13::ScaledDotProductAttention::validate_and_infer_types() {
     const auto& causal = get_causal();
     if (input_size >= 4 && !causal) {
         const auto& attention_type = get_input_element_type(3);
-        NODE_VALIDATION_CHECK(
-            this,
-            attention_type.is_real() || attention_type == element::boolean || attention_type.is_dynamic(),
-            "The element type of attention_mask must be either floating-point or boolean.");
+        NODE_VALIDATION_CHECK(this,
```
Comment (praasz) [inline src/core/src/op/scaled_dot_product_attention.cpp:47]: It is not according operator's specification as the input should be any supported floating-point type.

@mitruska 
Should be specification updated? Is possible to change the input this input type?

**Review by mitruska — COMMENTED** [review]
Looks like those changes are not currently covered by unit tests, can it be added for BOOL/U8 attention_mask at least for GPU? 

As `ScaledDotProductAttention` is a common operation, to extend it with U8 type support for "attention_mask" input, the **specification** should be updated, the **transformations** like SDPA decomposition, **reference** implementation and **other plugins**. It can be hard to cover all of the changes in single PR, so it should be planned accordingly.

We have discussed the implication of such change and possible solutions within the mentioned ticket (171516).

File: src/core/src/op/scaled_dot_product_attention.cpp (hunk 1 above)
Comment (mitruska) [reply src/core/src/op/scaled_dot_product_attention.cpp:47]: As `ScaledDotProductAttention` is a common operation, to extend it with U8 type support for "attention_mask" input, the **specification** should be updated, the **transformations** like SDPA decomposition, **reference** implementation and **other plugins**. It can be hard to cover all of the changes in single PR, so it should be planned accordingly.

We have discussed the implication of such change and possible solutions within the mentioned ticket (171516).
